DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -1463,7 +1463,7 @@ private void GameBroadcastChannel_CTCPReceived(object sender, ChannelCTCPEventAr
                 return;
 
             string msg = e.Message.Substring(5); // Cut out GAME part
-            string[] splitMessage = msg.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+            string[] splitMessage = msg.Split(new char[] { ';' });
```

Comment:
Have you verified it impacts anything else other than the changed code?

Reply:
I have removed two `StringSplitOptions.RemoveEmptyEntries` (one for CnCNet and the other for LAN) after reading codes from `BroadcastPlayerOptions()`. 

Test:
I tested on my PC with a VM, making sure the game option is correctly parsed without errors.

Theory:
1. Removing `RemoveEmptyEntries` option only affects the case there is an empty string between two separators. There should be no such cases.
2. Both `BroadcastPlayerOptions()` (one for CnCNet and the other for LAN) utilizes `ExtendedStringBuilder` to append strings with a separator `;`. Even if the string is empty, the extended string builder will normally append a separator. However, when the message is once again split, `RemoveEmptyEntries` causes empty strings be removed from the result array. We can observe that `splitMessage.Length` as well as array indexes are fixed magic numbers that correspond exactly with the order of `sb.Append()` in `BroadcastPlayerOptions()`. Thus I believe introducing `RemoveEmptyEntries` should be a mistake.

Reply:
The location of two `BroadcastPlayerOptions()`:
- https://github.com/CnCNet/xna-cncnet-client/blob/ac044f79a816a182051eb7c531ea1aae349c8c9b/DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs#L807C33-L807C55
- https://github.com/CnCNet/xna-cncnet-client/blob/ac044f79a816a182051eb7c531ea1aae349c8c9b/DXMainClient/DXGUI/Multiplayer/GameLobby/LANGameLobby.cs#L483